### PR TITLE
remove WaitSchedule() from etcdserver tests

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: pkg/testutil/recorder.go [hunk 1]
```diff
@@ -14,27 +14,112 @@
...
+
+func (r *recorderStream) Chan() <-chan Action {
+	return r.ch
+}
+
+func NewRecorderStream() Recorder {
+	return &recorderStream{ch: make(chan Action)}
+}
+
+func (r *recorderStream) Wait(n int) ([]Action, error) {
```
Comment (xiang90) [inline pkg/testutil/recorder.go]: I am OK with blocking and waiting here. But can we add a seconds level timeout? So we can fail fast?

Comment (heyitsanthony) [inline pkg/testutil/recorder.go]: Sure

**Comment by heyitsanthony** [comment]
updated-- ptal /cc @xiang90

**Comment by xiang90** [comment]
@heyitsanthony 

I am not convinced to move all nop implementations together with a real implementations. But it is definitely better putting them all in server_test.go.

LGTM

**Comment by xiang90** [comment]
There are a few back and forth in the last two commits. Ideally we should fix them.

**Comment by heyitsanthony** [comment]
ok back-and-forth cleaned up afaik

**Comment by xiang90** [comment]
LGTM

**Pull request merged**